src/rabbitmq.js

```diff
@@ -0,0 +1,356 @@
+  username: string;
+
+  constructor (opts) {
+    opts = assign({}, opts)
+    this.hostname = opts.hostname ||
+      process.env.RABBITMQ_HOSTNAME ||
+      'localhost'
+    this.port = opts.port || parseInt(process.env.RABBITMQ_PORT, 10) || 5672
+    this.username = opts.username || `${process.env.RABBITMQ_USERNAME}`
+    this.password = opts.password || `${process.env.RABBITMQ_PASSWORD}`
+    this.log = logger.child({
+      module: 'ponos:rabbitmq',
+      hostname: this.hostname,
+      port: this.port
+    })
+    this.subscriptions = new Immutable.Map()
+    this.subscribed = new Immutable.Set()
+    this.consuming = new Immutable.Map()
+    return this
+  }
+
+  /**
+   * Connect to the RabbitMQ server.
+   *
+   * @return {Promise} Promise that resolves once connection is established.
+   */
+  connect (): Promise {
+    if (this.connection || this.channel) {
+      return Promise.reject(new Error('cannot call connect twice'))
+    }
+    let authString = ''
+    if (this.username && this.password) {
+      authString = `${this.username}:${this.password}@`
+    }
+    const url = `amqp://${authString}${this.hostname}:${this.port}`
+    this.log.info({ url: url }, 'connecting')
+    return Promise
+      .resolve(amqplib.connect(url, {}))
+      .catch((err) => {
+        this.log.error({ err: err }, 'an error occured while connecting')
+        throw err
+      })
+      .then((conn) => {
+        this.connection = conn
+        this.log.info('connected')
+        this.connection.on('error', this._connectionErrorHandler.bind(this))
+
+        this.log.info('creating channel')
+        return Promise.resolve(this.connection.createChannel())
+          .catch((err) => {
+            this.log.error({ err: err }, 'an error occured creating channel')
+            throw err
+          })
+      })
+      .then((channel) => {
+        this.log.info('created channel')
+        this.channel = channel
+        this.channel.on('error', this._channelErrorHandler.bind(this))
+      })
+  }
+
+  /**
+   * Subscribe to a specific direct queue.
+   *
+   * @param {String} queue Queue name.
+   * @param {Function} handler Handler for jobs.
+   * @return {Promise} Promise that is resolved once queue is subscribed.
+   */
+  subscribeToQueue (queue: string, handler: Function): Promise {
+    const log = this.log.child({
+      method: 'subscribeToQueue',
+      queue: queue
+    })
+    log.info('subscribing to queue')
+    if (!this._isConnected()) {
+      return Promise.reject(new Error('you must .connect() before subscribing'))
+    }
+    if (!isFunction(handler)) {
+      log.error('handler must be a function')
+      return Promise.reject(
+        new Error(`handler for ${queue} must be a function`)
+      )
+    }
+    if (this.subscribed.has(`queue:::${queue}`)) {
+      log.warn('already subscribed to queue')
+      return Promise.resolve()
+    }
+    return Promise
+      .resolve(this.channel.assertQueue(queue, { durable: true }))
```

Comment:
can you opts here passed in? IIRC there are some repos that pass special ops
https://github.com/Runnable/hermes/blob/af660962b3bc67ed39811e17a4362a725442ff81/lib/event-jobs.js#L128